extensions/arc/deployment/src/test/java/io/quarkus/arc/test/producer/NullProducerMethodTest.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.arc.test.producer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.IllegalProductException;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import io.quarkus.test.QuarkusUnitTest;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class NullProducerMethodTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(NullProducerMethodTest.class, NullProducerMethodTest.MyBean.class,
+                            NullProducerMethodTest.MyBeanProducer.class));
+
+    @Inject
+    Instance<MyBean> instance;
+
+    @Test
+    public void testNullProducerShouldThrowIllegalProductException() {
+        assertThrows(IllegalProductException.class, instance::get);
+    }
+
+    class MyBean {
+    }
+
+    @ApplicationScoped
+    static class MyBeanProducer {
+        @Produces
+        MyBean produceNull() {
```

Comment:
In fact, the producer must be annotated with a normal scope. In this case, the producer bean is `@Dependent` and so returning `null` is legal.